contracts/strategy/rysk/RyskStrategy.sol

```diff
@@ -0,0 +1,234 @@
+ */
+contract RyskStrategy is BaseStrategy {
+    using SafeERC20 for IERC20;
+    using PercentMath for uint256;
+
+    // rysk liquidity pool that this strategy is interacting with
+    IRyskLiquidityPool public immutable ryskLqPool;
+    // pending withdrawal receipt
+    IRyskLiquidityPool.WithdrawalReceipt pendingWithdrawal;
+
+    /**
+     * Emmited when a withdrawal has been initiated.
+     *
+     *@param shares to be withdrawn
+     */
+    event RyskWithdrawalInitiated(uint256 shares);
+
+    // rysk liquidity pool cannot be 0 address
+    error RyskLiquidityPoolCannotBe0Address();
+    // no withdrawal initiated
+    error RyskNoWithdrawalInitiated();
+    // cannot complete withdrawal in the same epoch
+    error RyskCannotComipleteWithdrawalInSameEpoch();
+
+    /**
+     * @param _vault address of the vault that will use this strategy
+     * @param _admin address of the administrator account for this strategy
+     * @param _ryskLiquidityPool address of the rysk liquidity pool that this strategy is using
+     * @param _underlying address of the underlying token
+     */
+    constructor(
+        address _vault,
+        address _admin,
+        address _ryskLiquidityPool,
+        IERC20 _underlying
+    ) BaseStrategy(_vault, _underlying, _admin) {
+        if (_ryskLiquidityPool == address(0))
+            revert RyskLiquidityPoolCannotBe0Address();
+
+        ryskLqPool = IRyskLiquidityPool(_ryskLiquidityPool);
+
+        underlying.safeIncreaseAllowance(_ryskLiquidityPool, type(uint256).max);
+    }
+
+    //
+    // IStrategy
+    //
+
+    /**
+     * Rysk strategy is asynchronous meaning it doesn't support immediate withdrawals.
+     *
+     * @return false always
+     */
+    function isSync() external pure override(IStrategy) returns (bool) {
+        return false;
+    }
+
+    /// @inheritdoc IStrategy
+    function hasAssets()
+        external
+        view
+        virtual
+        override(IStrategy)
+        returns (bool)
+    {
+        return _getTotalShares() != 0;
+    }
+
+    /// @inheritdoc IStrategy
+    function investedAssets()
+        external
+        view
+        virtual
+        override(IStrategy)
+        returns (uint256)
+    {
+        return _sharesToUnderlying(_getTotalShares());
+    }
+
+    /// @inheritdoc IStrategy
+    function invest() external virtual override(IStrategy) onlyManager {
+        uint256 balance = underlying.balanceOf(address(this));
+
+        if (balance == 0) revert StrategyNoUnderlying();
+
+        emit StrategyInvested(balance);
+
+        ryskLqPool.deposit(balance);
+    }
+
+    /// @inheritdoc IStrategy
+    function withdrawToVault(uint256 _amount)
+        external
+        virtual
+        override(IStrategy)
+        onlyManager
+    {
+        if (_amount == 0) revert StrategyAmountZero();
+
+        uint256 shares = _underlyingToShares(_amount);
+
+        _resetPendingWithdrawalIfEpochAdvanced();
```

Comment:
not sure i understand the point of this

Reply:
this was removed since it wasn't correct, thanks for pointing out